src/FamilyHubs.RequestForSupport.Core/ApiClients/ReferralClientService.cs

```diff
@@ -47,19 +60,87 @@ public async Task<PaginatedList<ReferralDto>> GetRequestsForConnectionByOrganisa
+        var url = $"api/organisationreferrals/{organisationId}?orderBy={orderBy}&isAssending={isAscending}&pageNumber={pageNumber}&pageSize={pageSize}&includeDeclined=false";
 
         var request = new HttpRequestMessage
         {
             Method = HttpMethod.Get,
             RequestUri = new Uri(Client.BaseAddress + url),
         };
-
+       
         using var response = await Client.SendAsync(request);
 
         response.EnsureSuccessStatusCode();
 
         return await JsonSerializer.DeserializeAsync<PaginatedList<ReferralDto>>(await response.Content.ReadAsStreamAsync(), options: new JsonSerializerOptions { PropertyNameCaseInsensitive = true })
                ?? new PaginatedList<ReferralDto>();
     }
+
+    public async Task<ReferralDto> GetReferralById(long referralId)
+    {
+        var url = $"api/referral/{referralId}";
+
+        var request = new HttpRequestMessage
+        {
+            Method = HttpMethod.Get,
+            RequestUri = new Uri(Client.BaseAddress + url),
+        };
+
+        using var response = await Client.SendAsync(request);
+
+        response.EnsureSuccessStatusCode();
+
+        return await JsonSerializer.DeserializeAsync<ReferralDto>(await response.Content.ReadAsStreamAsync(), options: new JsonSerializerOptions { PropertyNameCaseInsensitive = true }) ?? default!;
+    }
+
+    public async Task<List<ReferralStatusDto>> GetReferralStatuses()
+    {
+        var request = new HttpRequestMessage
+        {
+            Method = HttpMethod.Get,
+            RequestUri = new Uri(Client.BaseAddress + "api/referralstatuses"),
+        };
+
+        using var response = await Client.SendAsync(request);
+
+        response.EnsureSuccessStatusCode();
+
+        return await JsonSerializer.DeserializeAsync<List<ReferralStatusDto>>(await response.Content.ReadAsStreamAsync(), options: new JsonSerializerOptions { PropertyNameCaseInsensitive = true }) ?? new List<ReferralStatusDto>();
+    }
+
+    public async Task<string> UpdateReferral(ReferralDto referralDto)
+    {
+        var request = new HttpRequestMessage
+        {
+            Method = HttpMethod.Put,
+            RequestUri = new Uri(Client.BaseAddress + $"api/referrals/{referralDto.Id}"),
+            Content = new StringContent(Newtonsoft.Json.JsonConvert.SerializeObject(referralDto), Encoding.UTF8, "application/json"),
+        };
+
+        using var response = await Client.SendAsync(request);
+
+        response.EnsureSuccessStatusCode();
+
+        var stringResult = await response.Content.ReadAsStringAsync();
+        return stringResult;
+    }
+
+    public async Task<string> UpdateReferralStatus(long referralId, ReferralStatus referralStatus, string? reason = null)
+    {
+        var request = new HttpRequestMessage
+        {
+            // copilot's suggestion. seems good
+            //Method = HttpMethod.Put,
```

Comment:
Should this commented out code be removed

Reply:
Changed it to a todo. Not a biggie though, so we can leave it for now.